code/datums/components/corpse_description.dm

```diff
@@ -0,0 +1,36 @@
+/**
+ * Mapping component which adds examine text to a corpse that is removed on revival (in case for some reason someone revives your space ruin mob)
+ * For immersive environmental story telling
+ */
+/datum/component/corpse_description
+	/// What do we display on examine?
+	var/description_text = ""
+	/// What do we display if examined by a clown?
+	var/naive_description = ""
+
+/datum/component/corpse_description/Initialize(description_text = "", naive_description = "")
+	. = ..()
+	if (!isliving(parent))
+		return COMPONENT_INCOMPATIBLE
+	src.description_text = description_text
+	src.naive_description = naive_description
+
+/datum/component/corpse_description/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_ATOM_EXAMINE, PROC_REF(on_examined))
+	RegisterSignal(parent, COMSIG_LIVING_REVIVE, PROC_REF(on_revived))
```

Comment:
Make this component a more generic description-until-signal component and pass in the signal to qdel in response to as part of init.